File: ruby/basics/fizzbuzz.rb
#!/usr/bin/env ruby
# Print FizzBuzz for 1 through 100, choosing the label by matching the pair of remainders.

def label(n)
  case [n % 3, n % 5]
  in [0, 0] then 'FizzBuzz'
  in [0, _] then 'Fizz'
  in [_, 0] then 'Buzz'
  else n.to_s
  end
end

(1..100).each { |n| puts label(n) }

File: python/basics/fizzbuzz.py
#!/usr/bin/env python3
# Print FizzBuzz for 1 through 100, choosing the label with a match on the remainders.


def label(n: int) -> str:
    match (n % 3, n % 5):
        case (0, 0):
            return "FizzBuzz"
        case (0, _):
            return "Fizz"
        case (_, 0):
            return "Buzz"
        case _:
            return str(n)


for number in range(1, 101):
    print(label(number))
